crates/sui-framework/sources/hex.move

```diff
@@ -23,6 +26,36 @@ module sui::hex {
         r
     }
 
+    /// Decode hex into `bytes`
+    /// Takes a hex string (no 0x prefix) (e.g. b"0f3a")
+    /// Returns vector of `bytes` that represents the hex string (e.g. x"0f3a")
+    /// Hex string can be case insensitive (e.g. b"0F3A" and b"0f3a" both return x"0f3a")
+    /// Aborts if the hex string does not have an even number of characters (as each hex charater is 2 characters long)
+    /// Aborts if the hex string contains non-valid hex characters (valid charaters are 0 - 9, a - f, A - F)
+    public fun decode(hex: vector<u8>): vector<u8> {
+        let (i, r, l) = (0, vector[], vector::length(&hex));
+        assert!(l % 2 == 0, EInvalidHexLength); 
+        while (i < l) {
+            let decimal = (decode_byte(*vector::borrow(&hex, i)) * 16) + 
+                          decode_byte(*vector::borrow(&hex, i + 1));
+            vector::push_back(&mut r, decimal);
+            i = i + 2;
+        };
+        r
+    }
+    
+    fun decode_byte(hex: u8): u8 {
+        if (/* 0 .. 9 */ 48 <= hex && hex < 58) {
+            hex - 48
+        } else if (/* A .. F */ 65 <= hex && hex < 71) {
+            10 + hex - 65
+        } else if (/* a .. f */ 97 <= hex && hex < 103) {
+            10 + hex - 97
+        } else {
+            abort(ENotValidHexCharacter)
```

Comment:
This was my bad in my suggestion, but `abort` is an operator rather than a macro.

```suggestion
            abort ENotValidHexCharacter
```